### PR TITLE
fix: prevent overwriting materials when OverwriteExportedMaterials is false

### PR DESCRIPTION
**Pull request by schinkowski**
Fix for this: https://github.com/unity3d-jp/MeshSync/issues/902

### CODE REVIEW

**Review by seandillon92 — DISMISSED** [review]

File: Runtime/Scripts/BaseMeshSync.cs [hunk 1]
```diff
@@ -1041,9 +1043,8 @@ public abstract partial class BaseMeshSync : MonoBehaviour, IObservable<MeshSync
         if (dst == null || dst.material == null)
             return;
         
-        if (importerSettings.OverwriteExportedMaterials)
-            dst.ShouldApplyMaterialData = true;
-
+        dst.ShouldApplyMaterialData = isNewMaterial || importerSettings.OverwriteExportedMaterials;
```
Comment (sindharta-tanuwijaya) [inline Runtime/Scripts/BaseMeshSync.cs]: I am thinking we should probably make `ShouldApplyMaterialData` to a local variable along with this fix, seeing that we only use it inside this function.

File: Runtime/Scripts/BaseMeshSync.cs [hunk 2]
```diff
@@ -1014,6 +1014,8 @@ public abstract partial class BaseMeshSync : MonoBehaviour, IObservable<MeshSync
 
         MaterialHolder dst = m_materialList.Find(a => a.id == materialID);
 
+        bool isNewMaterial = dst == null || dst.material == null;
+        
         //if (invalid && creating materials is allowed)
         if ((dst == null || dst.material == null || dst.name != materialName) && importerSettings.CreateMaterials) {
             if (null == dst) {
```
Comment (sindharta-tanuwijaya) [inline Runtime/Scripts/BaseMeshSync.cs:1021]: Perhaps it's better to remove line 1031
`dst.ShouldApplyMaterialData = false;`

as well ?

Comment (schinkowski) [reply Runtime/Scripts/BaseMeshSync.cs:1021]: Yeah, removed now

File: Runtime/Scripts/BaseMeshSync.cs (hunk 1 above)
Comment (schinkowski) [reply Runtime/Scripts/BaseMeshSync.cs]: Done

**Review by sindharta-tanuwijaya — APPROVED** [review]

**Pull request merged**